otcextensions/osclient/vpc/v2/peering.py

```diff
@@ -0,0 +1,333 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+"""VPC Peering v2 action implementations"""
+import logging
+
+from osc_lib import utils
+from osc_lib import exceptions
+from osc_lib.command import command
+
+from otcextensions.i18n import _
+
+LOG = logging.getLogger(__name__)
+
+STATUS_CHOICES = [
+    'PENDING_ACCEPTANCE',
+    'REJECTED',
+    'EXPIRED',
+    'DELETED',
+    'ACTIVE'
+]
+
+
+def _update_vpc_info(vpcinfo):
+    if 'vpc_id' in vpcinfo:
+        vpcinfo['router_id'] = vpcinfo.pop('vpc_id')
+    if 'tenant_id' in vpcinfo:
+        vpcinfo['project_id'] = vpcinfo.pop('tenant_id')
+    return vpcinfo
+
+
+def set_attributes_for_print(peerings):
+    for peering in peerings:
+        setattr(peering,
+                'request_vpc_info',
+                _update_vpc_info(peering.request_vpc_info))
+        setattr(peering,
+                'accept_vpc_info',
+                _update_vpc_info(peering.accept_vpc_info))
+        yield peering
+
+
+def translate_response(func):
+    def new(self, *args, **kwargs):
+        response = func(self, *args, **kwargs)
+        setattr(response,
+                'request_vpc_info',
+                _update_vpc_info(response.request_vpc_info))
+        setattr(response,
+                'accept_vpc_info',
+                _update_vpc_info(response.accept_vpc_info))
+        columns = (
+            'id',
+            'name',
+            'request_vpc_info',
+            'accept_vpc_info',
+            'description',
+            'created_at',
+            'updated_at',
+            'status'
+        )
+        data = utils.get_item_properties(response, columns)
+        return (columns, data)
+    new.__name__ = func.__name__
+    new.__doc__ = func.__doc__
+    return new
+
+
+class ListVpcPeerings(command.Lister):
```

Comment:
the output sucks a bit due to the not flattened structure. it would be much better if we can generate an output which as the local vpc id seperately listed from project id and also the peering vpc id

Reply:
Updated the output format